File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hm = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int x = target - nums[i];
            if(hm.containsKey(x)){
                return new int[]{hm.get(x), i};
            } else{
                hm.put(nums[i], i);
            }
        }
        return new int[]{0,0};
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let mp = new Map();
    for(let i = 0; i < nums.length; i++){
        let x = target - nums[i];
        if(mp.has(x)){
            return [mp.get(x), i];
        } else{
            mp.set(nums[i], i);
        }
    }
    return [];
};